test/vmtests/vmtests/test_no_change.py

```diff
@@ -69,6 +87,8 @@ def test_no_change(
 
     # Wait for VM to boot by waiting for it to request an IP address from the DHCP server.
     vm_ip_address = vm.get_vm_ip_address(timeout=30)
+    # iso booting takes longer due to the copying of artifacts to memory
+    time.sleep(30)
```

Comment:
Do not add arbitrary sleep calls. Add a wait loop of some kind. In this case, you can wait until the SSH port is open.

If you spin on the `SshClient` call, then make sure you are checking for a specific error. Alternatively, you could use Python's [socket](https://docs.python.org/3/library/socket.html) API to check if the TCP port is open yet.